tracing/tracing.go

```diff
@@ -0,0 +1,97 @@
+package tracing
+
+import (
+	"context"
+
+	"github.com/distribution/distribution/v3/configuration"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+)
+
+// InitOpenTelemetry reads config and initializes otel middleware, sets the exporter
+// propagator and global tracer provider
+func InitOpenTelemetry(config *configuration.Configuration) (func(), error) {
+	ctx := context.Background()
+
+	// Check if tracing is configured
+	if config.OpenTelemetry == nil {
+		logrus.Info("OpenTelemetry configuration not found, tracing is disabled")
+		return nil, nil
+	}
+
+	// Validate configuration
+	if err := config.OpenTelemetry.Validate(); err != nil {
+		return nil, errors.Wrap(err, "invalid open telemetry configuration")
```

Comment:
Any specific reason to use errors.wrap here, vs fmt.Errorf? I know that the package provides stack traces.

Reply:
IIRC `errors.Wrap` returns an error with a stack trace at the point it is called. stdlib wrapping doesnt give you that (just speculating here on behalf of the author )

Reply:
https://stackoverflow.com/questions/61933650/whats-the-difference-between-errors-wrapf-errors-errorf-and-fmt-errorf

I think this answers the question.